### PR TITLE
WT-8932 Call wait on forked child from Python tests and add extra debugging for PIDs

### PR DESCRIPTION
**Pull request by mickgraham-mongodb**
Call wait on forked child from Python tests to prevent defunct processes hanging around. Updated output of python tests to have prefix of pid (including if its the child) to help with debugging of timeout failures.

### CODE REVIEW

File: test/3rdparty/concurrencytest-0.1.2/concurrencytest.py
Comment (agorrod) [inline test/3rdparty/concurrencytest-0.1.2/concurrencytest.py:48]: It's possible that this would be the first time we'd diverged from the third party code - if that's the case, we should think carefully about it (i.e: ensure we don't loose the change if we ever decided to upgrade to a new version), and probably attempt to send the change upstream as well.

File: test/3rdparty/concurrencytest-0.1.2/concurrencytest.py
Comment (agorrod) [inline test/3rdparty/concurrencytest-0.1.2/concurrencytest.py:51]: Does this add support for parallel processing on Windows via the concurrency library? We've worked around that for years..
See WT-3656 and WT-2815 (I didn't look deeply, just saw the change in this file and comment about Windows)

**Review by agorrod — COMMENTED** [review]
I've got a couple of questions, and didn't test this change locally, but it does look good to me.

Don is *much* more proficient in Python than I am, so you should also wait for his review.

File: test/3rdparty/concurrencytest-0.1.2/concurrencytest.py
Comment (mickgraham-mongodb) [reply test/3rdparty/concurrencytest-0.1.2/concurrencytest.py:48]: Right, I'm not sure what our process for this is. Hoping Don can provide some guidance. I worked at it to limit the changes to this file only in terms of 3rd party changes. We did see this project (concurrencytest) is not active.

File: test/3rdparty/concurrencytest-0.1.2/concurrencytest.py
Comment (mickgraham-mongodb) [reply test/3rdparty/concurrencytest-0.1.2/concurrencytest.py:51]: No it doesn't. Our current solution on windows is to create buckets of unit-tests and run them separately. This works well there for now. My comment here is more for future changes if this is used on Windows (based on the python documentation how Windows requires a PID and we can't use -1 as all children).

**Review by agorrod — APPROVED** [review]
LGTM - thanks Mick

File: test/3rdparty/concurrencytest-0.1.2/concurrencytest.py
Comment (ddanderson) [reply test/3rdparty/concurrencytest-0.1.2/concurrencytest.py:48]: Yes, I'm pretty sure this is the first time we've diverged from Python third party code.  To make it clear to future maintainers, I'd do three things.  First, I'd rename `test/3rdparty/[concurrencytest-0.1.2/` to be something like `test/3rdparty/[concurrencytest-0.1.2-locally-modified/` .  The `run.py` doesn't care what the directory names are, so I believe this will continue to work everywhere.  Second, I'd throw a 1 line README in this directory, to make sure it catches anybody's attention.  Third, I'd put a comment here (in this file) alerting to the modification (sure it's in git, but....).

**Review by ddanderson — APPROVED** [review]
LGTM - some smallish comments/suggestions.

File: test/3rdparty/concurrencytest-0.1.2/concurrencytest.py
Comment (mickgraham-mongodb) [reply test/3rdparty/concurrencytest-0.1.2/concurrencytest.py:64]: We use no hang so its not a blocking wait, need the sleep. Will add a comment.

File: test/3rdparty/concurrencytest-0.1.2/concurrencytest.py
Comment (mickgraham-mongodb) [reply test/3rdparty/concurrencytest-0.1.2/concurrencytest.py:50]: Updated in latest push.

**Review by agorrod — APPROVED** [review]
LGTM - thanks Mick

**Review by ddanderson — APPROVED** [review]
LGTM!

**Review by agorrod — APPROVED** [review]
lgtm

**Pull request merged**
